models/lenet/lenet.hpp

```diff
@@ -0,0 +1,210 @@
+/**
+ * @file lenet.hpp
+ * @author Eugene Freyman
+ * @author Daivik Nema
+ * @author Kartik Dutt
+ * 
+ * Definition of LeNet generally used for object detection.
```

Comment:
I'm wondering if we should split this up into multiple PR's makes it easier to review, let me know what you think.

Reply:
Yes I agree with @zoq This PR is already quite big. 
Let us know what you think @kartikdutt18

Reply:
Agreed, as discussed on the IRC, I will remove the portion related to LeNet and object detection and keep only the data loader portion with its utility functions. I hope that makes sense. Let me know what you think.